### PR TITLE
ADR decisions for GUI

### PR DESCRIPTION
**Pull request by ebbaoliveberg**
## Read this before submitting the PR

1. Always create a Draft PR first
2. Go through sections 1-5 below, fill them in and check all the boxes
3. Make sure that the branch is updated; if there's an "Update branch" button at the bottom of the PR, rebase or update branch.
4. When all boxes are checked, information is filled in, and the branch is updated: mark as Ready For Review and tag reviewers (top right)
5. Once there is a submitted review, implement the suggestions (if reasonable, otherwise discuss) and request an new review.

If there is a field which you are unsure about, enter the edit mode of this description or go to the [PR template](../.github/pull_request_template.md); There are invisible comments providing descriptions which may be of help.

## 1. Description / Summary

This PR adds the GUI-related decisions taken so far as ADRs. 

## 2. Jira task / GitHub issue

**Is this a GitHub issue?** --> Add the link to the github issue

**Is this from a Jira task?** --> If your branch does not contain info regarding the Jira task ID, put it here.

HMS-2419
HMS-2432

## 3. Type of change - Add label

**What _type of change(s)_ does the PR contain? For an explanation of the different options below, enter edit mode of this PR description template.**

- `type: breaking`: Changes in this PR will cause existing functionality to not work as expected. The master branch of the API will no longer work with the CLI dev branch (and vice versa).
- `type: feature`: You've added new functionality or updated an existing one.
- `type: bug`: The PR fixes a bug.
- `type: docs`: The PR _only_ updates documentation.
- `type: dependency`: You've updated a dependency version, e.g. a python package (in requirements.txt).
- `skip-changelog`: None of the above mentioned labels fit in. E.g. a new GitHub Action, a PR containing _only_ tests, etc.

## 4. Additional information

- [x] I have added an entry to the [Sprintlog](../SPRINTLOG.md) 
- [ ] This is a PR to the `master` branch: _If checked, read [the release instructions](../docs/procedures/new_release.md)_ 
  - [ ] I have followed steps 1-9. 

## 5. Actions / Scans

**Make sure that the following checks/actions have passed.**

- **Black**

- **Prettier**

- **Pylint**

- **Yamllint**

- **Tests**

- **CodeQL**

- **Trivy**

- **Snyk**

- **TestPyPI**

If an action does not pass and you need help with how to solve it, enter edit mode of this PR template or go to the [PR template](../.github/pull_request_template.md).

### CODE REVIEW

**Comment by i-oden** [comment]
@i-oden Is reviewing this.

**Review by i-oden — CHANGES_REQUESTED** [review]
Review of `0001` first: Looks good, just want to add a bit more information.

**Review by i-oden — COMMENTED** [review]
`0001` approved ✅ 
Reviewing `0003`

File: docs/architecture/decisions/0003-omit-specific-component-tests.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Currenly, the complex components are not resued between pages. The test converage would be the same in both alternatives.
+
+## Decision
+
+The component specific test will be omitted.
+
+## Consequences
+
+As the complex components are not resued between pages at the moment, duplicate testing schemas would not increase the readability of the test coverage of the componet. If the complex components are used in more than one page, we should bring this point up for discussion and reevaluate oour decision.
```
Comment (i-oden) [inline docs/architecture/decisions/0003-omit-specific-component-tests.md:27]:
```suggestion
As the complex components are not reused between pages at the moment, duplicate testing schemas would not increase the readability of the test coverage of the component. If the complex components are used in more than one page, we should bring this point up for discussion and reevaluate our decision.
```

**Review by i-oden — CHANGES_REQUESTED** [review]
Review of `0003`:
Commented on a few typos. In general I think it looks good but I'm not 100% sure that omitting the tests is the way to go. Would it be possible to add a couple examples of what components are already tested by textual and which components (at the time of writing) would be tested by us specifically?

File: docs/architecture/decisions/0002-use-project-centered-layout.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Initial mockups featured a menu based layout, with menu items corresponding to the differend CLI group commands, but was decided against as all core features included project selection, hence a project centered one page layout was suggested.
+
+## Decision
+
+We will use a project centered layoud for the DDS GUI.
+
+## Consequences
+
+The GUI will be easier to navigate for end users and focus on the core features for these types of users. For administrative use, the project centered layout is not the most intuitive, so initially the CLI will be used for these operations. However, it would be possible to add new modules (project mode, admin mode) fot including administrative users in the GUI.
```
Comment (i-oden) [inline docs/architecture/decisions/0002-use-project-centered-layout.md:24]:
```suggestion
The GUI will be easier to navigate for end users and focus on the core features for these types of users. For administrative use, the project centered layout is not the most intuitive, so initially the CLI will be used for these operations. However, it would be possible to add new modules (project mode, admin mode) for including administrative users in the GUI.
```

**Review by i-oden — COMMENTED** [review]
Approved `0002` ✅

File: docs/architecture/decisions/0003-omit-specific-component-tests.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+The pattern for UI components in the DDS GUI is to implement the Textual componet class. This is to ensure uniformity over the application with consitent component styling and functionallity.
+Mostly, the UI components only feature styling (for example buttons and containers), but some feature more complex funtionallity (for example modals and tree views). For the simpler UI componetns, no additionalal tests are needed as they are covered by the frameworks testsing. For the more complex components with added logic, these need additional testing.
+
+There are two options for the testing of these components:
+
+1. Test the UI components more thurughly through integration test corresponding to the different userflows, or
+2. Test the UI components in the integration tests **AND** with designated component tests.
+
+Currenly, the complex components are not resued between pages. The test converage would be the same in both alternatives.
+
```
Comment (i-oden) [inline docs/architecture/decisions/0003-omit-specific-component-tests.md:20]: I don't exactly know how the component tests would look, but I agree that there would probably be a bit of code duplication. I think we should add some of your reply in the comment to the doc. 

```suggestion
The following is an example of one of the "simpler" components. In this case the component only changes some of the styling of the parent class `Button` (from Textual) and therefore implementing a custom test for the functionality of the `DDSButton` would not give us anything. 

<code>
class DDSButton(Button):
    """Regular button widget with uppercase title.
    Args:
        label: The label to be displayed on the button.
    """

    def __init__(self, label: str, *args: Any, **kwargs: Any) -> None:
        super().__init__(label.upper(), *args, **kwargs)

    DEFAULT_CSS = """
    DDSButton {
        padding: 0 2 0 2;
    }
    DDSButton.wide {
        width: 100%;
    }
    """
</code>

For the more complex components, such as `DDSTreeView`, where the goal is to display the project contents in a "tree view" (thus something specifically implemented by us), it might make more sense to add component tests. It is however only planned to be used in one place at the moment, so its not an actual "reusable component", it is only treated as such for code readability and consistency. Adding "strict" component tests and integration tests would in this case produce partly duplicate test code. 
```

Comment (i-oden) [reply docs/architecture/decisions/0003-omit-specific-component-tests.md]: @ebbaoliveberg What do you think?

Comment (i-oden) [reply docs/architecture/decisions/0003-omit-specific-component-tests.md:43]: Great! I'll tag the team as well to see what they think 
@valyo @rv0lt

File: docs/architecture/decisions/0003-implement-specific-component-tests.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+Date: 2025-08-04
+
+## Status
+
+Accepted
+
+## Context
+
+The pattern for UI components in the DDS GUI is to implement the Textual component class. This is to ensure uniformity over the application with consistent component styling and functionality.
+Mostly, the UI components only feature styling (for example buttons and containers), but some feature more complex functionality (for example modals and tree views). For the simpler UI components, no additional tests are needed as they are covered by the frameworks testing. For the more complex components with added logic, these need additional testing.
```
[the remark this answers is not in the document]
Comment (ebbaoliveberg) [reply docs/architecture/decisions/0003-implement-specific-component-tests.md:12]: I was thinking "some feature" referring to the components, but the second part is more clear now! I will change it.

[the remark this answers is not in the document]
Comment (ebbaoliveberg) [reply docs/architecture/decisions/0003-implement-specific-component-tests.md]: Added clarification in the first part.

**Review by i-oden — CHANGES_REQUESTED** [review]
I approve all of the ADRs, just add an entry to the sprintlog and I'll approve the PR (I'm pretty sure I saw a message somewhere where you asked about this but I don't know where cause I can't find it now)

File: SPRINTLOG.md [hunk 5]
```diff
@@ -441,3 +441,7 @@ _Empty sprint_
 # Summer 2025
 
 - Refactoring the authentication code to work for both the cli and the gui ([#762](https://github.com/ScilifelabDataCentre/dds_cli/pull/762))
+
+# 2025-08-4 - 2025-08-15
```
Comment (i-oden) [inline SPRINTLOG.md:445]:
```suggestion
# 2025-08-04 - 2025-08-15
```

**Review by i-oden — APPROVED** [review]
Looks good!

**Comment by i-oden** [comment]
@valyo Do you want to weigh in on the component testing (`0003`)? If not we can dismiss your review.
